Checkrelease snippet for Vagrantfile.  This prevents folks from deplo…

jayunit100 opened the pull request:
@mikedanese @derekwaynecarr  ... here's a quick way to do it.  Let me know if you have  a better way.
fixes #18079

jayunit100 committed:
Vagrantfile.  This prevents folks from deploying salt configs that don't
match whats in the binary, or at least warns them.  Also generalizes
error message behaviour.

mikedanese commented on Vagrantfile:
```diff
@@ -7,16 +7,52 @@ VAGRANTFILE_API_VERSION = "2"
+end
+
+def binarycheck
+  if ARGV.first == "up"
```
if is unnecessary given previous check.

jayunit100 commented:
Ok @mikedanese 
- Updated PR according to conversation in #18079 (env var + warning).
- cleaned error checking behaviour so its all done w/ functions.

mikedanese commented on Vagrantfile:
```diff
@@ -7,16 +7,52 @@ VAGRANTFILE_API_VERSION = "2"
+    
+    # make sure there is a release...
+    err="No kube release found ! If running from a clone of the git repo, please run 'make quick-release'."
+    return false, err if ! File.exist?(version)
```
This will always fail if I download a release tar using https://get.k8s.io/ or the github releases tab.

jayunit100 replied:
Ohhhh I forgot about that use case.  Any clever generalized ideas on what to look for?

jayunit100 replied:
ah i see, there is a good way to check  in provision-master.sh

mikedanese replied:
Please don't push this into provisioning code. Can you factor this to not be kube-up generic?

mikedanese commented on Vagrantfile:
```diff
@@ -7,16 +7,52 @@ VAGRANTFILE_API_VERSION = "2"
+    # make sure release matches source...
+    gitcommit = `git log -1 --abbrev-commit --format="%h"`.gsub("\n","")
+    releasecommit = `cat #{version}`
+    checkbinary = (ENV['IGNORE_BINARY_CHECK']  == nil )
```
non-descriptive env var name.

mikedanese commented:
Why is this vagrant specific? Shouldn't this be factored to work with all kube-ups?

jayunit100 commented:
All kube-ups ? I kinda thought  so also,   but.. what about gke/juju/ ?

mikedanese commented:
if the enter through ./cluster/kube-up there shouldn't be a problem, right?

jayunit100 commented:
well, do they even need a release at all ?
i guess even if they dont, they can override the check .
the majority of kube-ups do need a release i assume

jayunit100 commented:
seeing as how its going to be more complicated for other tools (i.e. gce, non dev releases), and that this is a minor improvement, im  closing for now...  

I think in the future maybe there will be a salt/ansible/go solution to this (i.e. when server spins up, send it a version arg or something).
